FastSimulation/SimplifiedGeometryPropagator/src/ParticleManager.cc

```diff
@@ -251,22 +251,25 @@ std::unique_ptr<fastsim::Particle> fastsim::ParticleManager::nextGenParticle()
                 
         // particles which do not descend from exotics must be produced within the beampipe
         int exoticRelativeId = 0;
-        if (productionVertex->position().perp2() * lengthUnitConversionFactor2_ > beamPipeRadius2_)  //
+        const bool producedWithinBeamPipe = productionVertex->position().perp2() * lengthUnitConversionFactor2_ < beamPipeRadius2_;
+        if (!producedWithinBeamPipe)  //
         {
             exoticRelativesChecker(productionVertex, exoticRelativeId, 0);
-            if (!isExotic(exoticRelativeId)) {
+            if (!isExotic(fixLongLivedBug_, exoticRelativeId)) {
                 continue;
             }
-        }
-                  
-                  
-        // particle must not decay before it reaches the beam pipe
-        if(endVertex && endVertex->position().perp2()*lengthUnitConversionFactor2_ < beamPipeRadius2_)
+        }	
+        const bool decayedWithinBeamPipe = endVertex && endVertex->position().perp2() * lengthUnitConversionFactor2_ < beamPipeRadius2_;
+        if(decayedWithinBeamPipe)
         {
             continue;
         }
 
-
+        // SM particles that descend from exotics and cross the beam pipe radius should make hits but not be decayed, by default it will duplicate FastSim hits for long lived particles and so anything produced without activating fixLongLivedBug_ is physically wrong
```

Comment:
Could you please split this comment line, so that it can be more easily readable?